src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -653,10 +655,11 @@ public Object onTraversalSuccess() {
                     if (goodModel.failedToConverge) {
                         if (outputModel != null) {
                             final GATKReport report = writeModelReport(goodModel, null, USE_ANNOTATIONS);
-                            try (final PrintStream modelReportStream = new PrintStream(outputModel)) {
+                            try (final PrintStream modelReportStream = new PrintStream(outputModel.getOutputStream())) {
+                                if (modelReportStream.checkError()) {
+                                    throw new UserException.CouldNotCreateOutputFile(outputModel, "I/O stream error writing to report output");
+                                }
                                 report.print(modelReportStream);
```

Comment:
You need to call `checkError()` after `print()`, since `print()` can also set the error flag.

Reply:
Also, as noted above the `PrintStream` constructor internally creates an `OutputStreamWriter`, which can throw an exception. We should catch that exception and wrap in a `UserException`

Reply:
Done - moved to a separate private method to consolidate the two call sites with this problem. Its still kind of awkward though - we really need to fix GATKReport (which is already included in the list in https://github.com/broadinstitute/gatk/issues/6610).